flint/flint-core/src/main/scala/org/opensearch/flint/core/metadata/FlintMetadata.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.flint.core.metadata;
+
+import java.util.Map;
+
+/**
+ * Flint metadata follows Flint index specification and defines metadata
+ * for a Flint index regardless of query engine integration and storage.
+ */
+public class FlintMetadata {
+
+  /** Field name and type for each field in a Flint index. */
+  private final Map<String, String> schema;
+
+  /** Meta info for the Flint index. */
+  private final Map<String, Object> meta;
+
+  public FlintMetadata(Map<String, String> schema,
+                       Map<String, Object> meta) {
+    this.schema = schema;
+    this.meta = meta;
+  }
+
+  public Map<String, String> getSchema() {
```

Comment:
is it possible return String? My idea is the schema specification is `json string`. Does it make sense to let the consumer decide to how to parse it.

Reply:
Sure, changed the metadata content to a single string for flexibility. Add TODO comment for defining the metadata format and adding strong-typed class later.